docs/03-es-module-syntax.md

```diff
@@ -8,6 +8,21 @@ The following is intended as a lightweight reference for the module behaviors de
 
 Imported values cannot be reassigned, though imported objects and arrays _can_ be mutated (and the exporting module, and any other importers, will be affected by the mutation). In that way, they behave similarly to `const` declarations.
 
+#### import.meta usage
+
+import.meta.url holds a reference to the currentl imported script and is often used to get the current url.
+
+```js
+// Assign importMetaUrl as it is only accessable on first instantiation cycle of the module
+const currentUrl = import.meta.url;
+
+const willFailCanNotAccessImportMetaUrl = async () => console.log(import.meta.url);
+const getUrl = async () => console.log(currentUrl);
```

Comment:
I am sorry but I do not understand this example at all. Why would accosting `import.meta.url` work in line 17 but not line 19? Why are the functions `async` if they do not do anything async? If `import.meta.url` is not supported, then the entire script will fail much earlier with a SyntaxError.

Reply:
@lukastaegert because in runtimes it is not there anymore after the first loop of the engine if you do not assign it directly on the first iteration. I do not even exactly know why that is the case but it is 100% the case at last in v8 and graaljs. It does only exist on instantiation and not at runtime after that.

Reply:
ok behavior changed over time with current node i could not do a minimal reproducer so it is ok i guess to close this.